config/di.php

```diff
@@ -1,5 +1,19 @@
 <?php
 return [
+    'event_manager' => \DI\decorate(function($eventManager, $container) {
+        $template = $contaner->get('template');
+        $eventManager->attach('ERROR_DISPATCH', function (\GianArb\Penny\Event\HttpFlowEvent $event) use ($template) {
+            $e = $event->getException();
+
+            $response = $event->getResponse()->withStatus($e->getCode());
+            $event->setResponse($response);
+
+            $response->getBody()->write($template->render("error/40x", [
+                'title' => $e->getMessage(),
+                'exception' => $e,
+            ]));
```

Comment:
sounds like it! That's a common mistake unfortunately :/ (I did it too) I'm still looking for a solution to help prevent that. The thing is that a decorator could return `null`, so throwing an exception if a decorator returns `null` isn't good… Ideas welcome!